cmd/kustomizepatch/runner.go

```diff
@@ -0,0 +1,204 @@
+		// Else:
+		// REQUIRES GitRepository pointing to giantswarm/config!
+		// 1. Check if sourceServiceEnvVar is a valid URL (format: http://<kubernetes service ref>)
+		// 1.5. (optional) Is it possible to check if we have the latest version pulled?
+		// 2. Download http://<sourceServiceEnvVar>/gitrepository/<namespace>/<name>/latest.tar.gz
+		//    example: http://<>/gitrepository/flux-system/gitrepository-giantswarm-config/latest.tar.gz
+		//    if: there is a GitRepository CR 'flux-system/gitrepository-giantswarm-config'
+		// 3. Untar with github.com/fluxcd/pkg/untar (untar.Untar())
+		// 4. Set dir to location of untarred files
+		dir = "/home/jakub/Work/giantswarm/config"
+
+		var installation string
+		{
+			installation = os.Getenv(installationEnvVar)
+			if installation == "" {
+				return nil, microerror.Maskf(invalidConfigError, "%q environment variable is required", installationEnvVar)
+			}
+		}
+
+		var vaultClient *vaultapi.Client
+		{
+			vaultClient, err = createVaultClientUsingEnv(ctx)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+		}
+
+		var gen *generator.Service
+		{
+			c := generator.Config{
+				VaultClient: vaultClient,
+
+				// TODO(kuba): r.config.Dir will be constant, probably. Remove it.
+				Dir:          dir,
+				Installation: installation,
+			}
+
+			gen, err = generator.New(c)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+		}
+
+		in := generator.GenerateInput{
+			App:       r.config.AppName,
+			Name:      addNameSuffix(r.config.Name),
+			Namespace: giantswarmNamespace,
+
+			ExtraAnnotations: map[string]string{
+				meta.Annotation.XAppInfo.Key():        meta.Annotation.XAppInfo.Val(r.config.AppCatalog, r.config.AppName, r.config.AppVersion),
+				meta.Annotation.XCreator.Key():        meta.Annotation.XCreator.Default(),
+				meta.Annotation.XInstallation.Key():   installation,
+				meta.Annotation.XProjectVersion.Key(): meta.Annotation.XProjectVersion.Val(false),
+			},
+			ExtraLabels: nil,
+		}
+
+		configmap, secret, err = gen.Generate(ctx, in)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var appCR *applicationv1alpha1.App
+	{
+		c := app.Config{
+			AppCatalog:          r.config.AppCatalog,
+			AppName:             r.config.AppName,
+			AppNamespace:        r.config.AppDestinationNamespace,
+			AppVersion:          r.config.AppVersion,
+			ConfigVersion:       configmap.Annotations[meta.Annotation.ConfigVersion.Key()],
+			DisableForceUpgrade: r.config.AppDisableForceUpgrade,
+			Name:                r.config.Name,
+		}
+
+		appCR = app.NewCR(c)
+
+		appCR.Spec.Config.ConfigMap = applicationv1alpha1.AppSpecConfigConfigMap{
+			Name:      configmap.Name,
+			Namespace: configmap.Namespace,
+		}
+		appCR.Spec.Config.Secret = applicationv1alpha1.AppSpecConfigSecret{
+			Name:      secret.Name,
+			Namespace: secret.Namespace,
+		}
+	}
+
+	output := []*kyaml.RNode{}
+	for _, item := range []runtime.Object{configmap, secret, appCR} {
```

Comment:
This is what's new. We don't print out the result, but marshal it into `kyaml.RNode` objects instead.

Reply:
What is the difference? I assumed kyaml would do the same as printing a normal yaml

Reply:
The output is not printed in this case. We pass `r.run` as argument to build a [processor function](https://github.com/giantswarm/konfigure/pull/68/files#diff-ddf6200dd9a24122d2dfb0aed6db33f25c0ce38b796dbc1afb96895e1313085dR46-R49) using `kustomize` libraries. The `func (items []*kyaml.RNode) ([]*kyaml.RNode, error)` format is required. The list of `kyaml.RNodes` is being read and used by `kustomize` after we return it.